### PR TITLE
Add org.featherwallet.Feather

### PR DESCRIPTION
**Pull request by vorot93**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/feather-wallet/feather/issues/47
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.featherwallet.Feather.yaml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+          /app/share/icons/hicolor/64x64/apps/org.featherwallet.Feather.png'
+      - 'install -D src/assets/images/appicons/96x96.png
+          /app/share/icons/hicolor/96x96/apps/org.featherwallet.Feather.png'
+      - 'install -D src/assets/images/appicons/128x128.png
+          /app/share/icons/hicolor/128x128/apps/org.featherwallet.Feather.png'
+      - 'install -D src/assets/images/appicons/256x256.png
+          /app/share/icons/hicolor/256x256/apps/org.featherwallet.Feather.png'
+
+
+finish-args:
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:138]: finish-args is usually found above modules.

File: org.featherwallet.Feather.yaml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  # Show windows through Wayland
+  - --socket=wayland
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:140]: so what happen if there is no wayland?

File: org.featherwallet.Feather.yaml [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+          --config Release
+          -j ${FLATPAK_BUILDER_N_JOBS}'
+      - 'cmake
+          --install build/release
+          --config Release'
+      - 'mv -v /app/feather /app/bin/feather'
+        # Metadata: .desktop file (needs slight patching)
+      - 'install -D src/assets/feather.desktop
+          /app/share/applications/org.featherwallet.Feather.desktop'
+      - "sed -i 's/Icon=feather/Icon=org.featherwallet.Feather/'
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:118]: please use `desktop-file-edit` instead.

File: org.featherwallet.Feather.yaml [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    sources:
+      - type: git
+        url: 'https://github.com/feather-wallet/feather'
+        tag: '2.4.4'
+        commit: ffd6d332daa2f50da3f434de35837204d973ca39
+      # Metadata: Appdata XML
+      - type: file
+        path: org.featherwallet.Feather.appdata.xml
+        dest: src/assets/
+    buildsystem: simple
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:92]: this should be buildsystem `cmake-ninja` (or `cmake` if the former doesn't work)

File: org.featherwallet.Feather.yaml [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    sources:
+      - type: git
+        url: 'https://github.com/jedisct1/libsodium.git'
+        tag: '1.0.18-RELEASE'
+        commit: 940ef42797baa0278df6b7fd9e67c7590f87744b
+    buildsystem: autotools
+
+
+  - name: feather
+    sources:
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:83]: usually soure are listed last in the module. This comment applies to the other modules too.

File: org.featherwallet.Feather.appdata.xml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+			<description>
+				<ul>
+					<li>Sweeping or sending all spendable balance to an integrated address should no longer throw an error</li>
+					<li>Fixed an issue that could cause the built-in updater to fail to install Feather (Windows installer)</li>
+					<li>Importing a transaction should no longer hang the UI</li>
+					<li>Updated Monero to v0.18.1.0</li>
+				</ul>
+			</description>
+		</release>
+	</releases>
```
Comment (hfiguiere) [inline org.featherwallet.Feather.appdata.xml:98]: missing mandatory `content_rating`.

File: org.featherwallet.Feather.yaml [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+      - '--without-gtk'
+      - '--without-python'
+      - '--without-dbus'
+    sources:
+      - type: archive
+        url: 'https://www.linuxtv.org/downloads/zbar/zbar-0.23.90.tar.gz'
+        sha256: ff857dd7e3dbe043dac3765b5182c91dfd0477800713a75d15287d797cee60fa
+
+  - name: qrencode
+    buildsystem: cmake
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:62]:
```suggestion
    buildsystem: cmake-ninja
```

File: org.featherwallet.Feather.yaml [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+    sources:
+      - type: git
+        url: 'https://github.com/fukuchi/libqrencode.git'
+        tag: v4.1.1
+        commit: 715e29fd4cd71b6e452ae0f4e36d917b43122ce8
+
+  - name: zip
+    buildsystem: cmake
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:70]:
```suggestion
    buildsystem: cmake-ninja
```

File: org.featherwallet.Feather.yaml [hunk 9]
```diff
@@ -0,0 +1,138 @@
+app-id: org.featherwallet.Feather
+runtime: org.kde.Platform
+runtime-version: '6.4'
+sdk: org.kde.Sdk
+command: feather
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:8]: --share=ipc is needed for x11 fallback too

**Comment by hfiguiere** [comment]
bot, build org.featherwallet.Feather

File: org.featherwallet.Feather.yaml [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+      - type: git
+        url: 'https://github.com/jedisct1/libsodium.git'
+        tag: '1.0.18-RELEASE'
+        commit: 940ef42797baa0278df6b7fd9e67c7590f87744b
+
+  - name: feather
+    buildsystem: cmake-ninja
+    builddir: true
+    config-opts:
+      - -D BUILD_TAG="linux-x64"
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:90]: what about aarch64?

File: org.featherwallet.Feather.yaml [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+      - -D REPRODUCIBLE=Off
+    post_install:
+      - 'mv -v /app/feather /app/bin/feather'
+    sources:
+      - type: git
+        url: 'https://github.com/feather-wallet/feather'
+        tag: '2.4.4'
+        commit: ffd6d332daa2f50da3f434de35837204d973ca39
+      - type: shell
+        commands:
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:104]: this should all be in post-install, not a shell source.

File: org.featherwallet.Feather.yaml [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+            /app/share/icons/hicolor/48x48/apps/org.featherwallet.Feather.png'
+        - 'install -D src/assets/images/appicons/64x64.png
+            /app/share/icons/hicolor/64x64/apps/org.featherwallet.Feather.png'
+        - 'install -D src/assets/images/appicons/96x96.png
+            /app/share/icons/hicolor/96x96/apps/org.featherwallet.Feather.png'
+        - 'install -D src/assets/images/appicons/128x128.png
+            /app/share/icons/hicolor/128x128/apps/org.featherwallet.Feather.png'
+        - 'install -D src/assets/images/appicons/256x256.png
+            /app/share/icons/hicolor/256x256/apps/org.featherwallet.Feather.png'
+  - name: extra-files
```
Comment (hfiguiere) [inline org.featherwallet.Feather.yaml:123]: you don't need this module. just fold it into the main one.

File: org.featherwallet.Feather.yaml (hunk 11 above)
Comment (hfiguiere) [reply org.featherwallet.Feather.yaml:104]: just that with `builddir` set to true files in post-install need to be referenced with `../` to be relative to the top source.

File: org.featherwallet.Feather.yaml (hunk 12 above)
Comment (hfiguiere) [reply org.featherwallet.Feather.yaml:123]: `post-install` is what it's for.

Comment (hfiguiere) [reply org.featherwallet.Feather.yaml:123]: but the right thing is to have it upstream. That's where it belongs.

**Comment by hfiguiere** [comment]
bot, build org.featherwallet.Feather

**Comment by barthalion** [comment]
Would be nice to get some acknowledgement from upstream.

**Pull request closed without merging**